src/api/firebase.js

```diff
@@ -81,36 +80,29 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
  * @param {Object} itemData fields in each document of the firebase collection
  */
 export async function updateItem(listId, id, itemData) {
-	let daysSinceLastPurchase;
-	let previousEstimate;
-
-	if (itemData.dateLastPurchased) {
-		previousEstimate = getDaysBetweenDates(
-			itemData.dateLastPurchased.toMillis(),
-			itemData.dateNextPurchased.toMillis(),
-		);
-
-		daysSinceLastPurchase = getDaysBetweenDates(
-			itemData.dateLastPurchased.toMillis(),
-		);
-	} else {
-		previousEstimate = getDaysBetweenDates(
-			itemData.dateCreated.toMillis(),
-			itemData.dateNextPurchased.toMillis(),
-		);
-
-		daysSinceLastPurchase = getDaysBetweenDates(
-			itemData.dateCreated.toMillis(),
-		);
-	}
+	const previousEstimate = itemData.daysSinceLastPurchase
+		? getDaysBetweenDates(
+				itemData.dateLastPurchased,
+				itemData.dateNextPurchased,
+		  )
+		: getDaysBetweenDates(itemData.dateCreated, itemData.dateNextPurchased);
+	console.log('previousEstimate', previousEstimate);
 
+	const daysSinceLastPurchase = itemData.daysSinceLastPurchase
+		? getDaysBetweenDates(itemData.dateLastPurchased)
+		: getDaysBetweenDates(itemData.dateCreated);
+	//console.log('days since last purchased ', daysSinceLastPurchase);
 	let updatePreviousEstimate = calculateEstimate(
 		previousEstimate,
 		daysSinceLastPurchase,
 		itemData.totalPurchases,
 	);
```

Comment:
If the item is purchased for the first time, perhaps you want to consider keeping the previous estimate instead of calculating it. Otherwise, the estimated date gets updated to the same day.

Reply:
Also, for naming, `updatePreviousEstimate` may be a good name for a function, but for a variable, it could be more descriptive. What is the value returned? Since calculateEstimate returns a whole number indicating the number of days until the user is likely to make their next purchase is there a better name that could be used?